electrum/wallet.py

```diff
@@ -1952,6 +1972,14 @@ def fee_estimator(size: Union[int, float, Decimal]) -> int:
                     amount = change[0].value
                     if amount <= self.lnworker.num_sats_can_receive():
                         tx.replace_output_address(change[0].address, DummyAddress.SWAP)
+
+            self.logger.debug(f'coinchooser returned tx with {len(tx.inputs())} inputs and {len(tx.outputs())} outputs')
+            to_be_consumed = sum(c.value_sats() for c in tx.inputs() if c in coins)
+            amount_retained = sum(o.value for o in tx.outputs() if self.is_mine(o.address))
+            to_be_spent = to_be_consumed - amount_retained
+            if to_be_spent > 0 and self.should_keep_reserve_utxo(to_be_spent, is_anchor_channel_opening):
+                raise NotEnoughFunds()
```

Comment:
could you add a unit test that enters this branch?

Reply:
test added